crates/schema-inference/src/snapshots/schema_inference__analyze__test__string_format_number_back_to_string.snap

```diff
@@ -0,0 +1,19 @@
+---
+source: crates/schema-inference/src/analyze.rs
+expression: schema
+---
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "required": [
+    "a"
+  ],
+  "properties": {
+    "a": {
+      "type": [
+        "integer",
+        "string"
```

Comment:
There is a note here worth mentioning:

Documents are:
```
{"a": "1"}
{"a": "blah"}
```

Since one of the documents has a number as its string value, we give it the type `string, integer` and `format: integer`.
After merging it with the next document that shows a non-numeric string, we remove `format`, but it's harder to remove the type `integer` since we don't know at this stage how was that type added in the first place: maybe we did actually see an `integer` value at some point during the inference, or maybe we didn't and we just added this type because we saw strings with numeric values.

To put it another way: while it is valid for a field to be `type: string, integer` and only include strings, it is not valid for a field to have value `"blah"` with `format: integer`, so we can address the latter case, but the first case is ambiguous for us.

I don't think this is a big deal, but worth mentioning. Let me know if you think this can be problematic.

Reply:
I was wondering about this. Is it possible to add some metadata to the inferred schema? Also agree that it's not super important right now

Reply:
> Since one of the documents has a number as its string value, we give it the type string, integer and format: integer.

This is a drive by, but: shouldn't we give it _only_ type `string`, as in `{type: string, format: integer}`? We have yet to see an actual integer.

Reply:
@jgraettinger I think in that instance we don't convert that field's values into a number, we only do it if the type permits it.

But it's valid to say unless the data itself contains values which are actually of type integer / number, we do not convert.

Reply:
Okay, thinking about this more, there will be another edge case if we decide to only give it type `string`:

Imagine these documents come in:
```
{"a": 1}
{"a": "2"}
{"a": "NaN"}
```

After reading the first two documents, schema inference will suggest:
```
a:
  type: [string, integer]
  format: integer
```

And on reading the third document, we will have:
```
a:
  type: [string, integer]
  format: number
```

But this is wrong, we would expect that on reading the `NaN` which is a special case of number, we would get:
```
a:
  type: [string, number]
  format: number
```

Reply:
I think the current implementation's caveat is something we can live with. Unless someone has a strong opinion against this, I will merge this PR tomorrow.

Reply:
Given the `1, "2", "NaN" sequence:

```
a:
  type: [string, integer]
  format: number
```
Seems 100% correct to me. We _haven't_ seen a JSON number, only a string-encoded number (as NaN, which is never an integer), making this is the most-correct and tightly-fitting schema.

Reply:
Ahh, do we not handle the `{type: [string, integer], format: number}` case downstream within the SQL materializations? Could we?

Reply:
@jgraettinger no we can't handle that `format` in this case. We can only do conversion if the conversion fits with the `type`. In this instance the materialized field will have type `string, integer` and the format is basically no-op from our perspective, since `NaN` cannot be materialized into an `integer` field.

Reply:
@jgraettinger From my perspective, `1, 2, "NaN"` is the same as `1, 2, 1.1` in this instance, so my expectation of the resulting type is different from yours